File: src/main/java/ru/trukhmanov/twochairsbackend/entity/UserAnswer.java
package ru.trukhmanov.twochairsbackend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name="user_answers")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class UserAnswer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name="user_id", nullable=false)
    private Long userId;

    @Column(name="question_id", nullable=false)
    private Long questionId;

    @Column(name="answer", nullable=false)
    private Character answer; // "A" or "B"
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/entity/UserAnswer.kt
package ru.trukhmanov.twochairsbackend.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "user_answers")
class UserAnswer(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "user_id", nullable = false)
    var userId: Long? = null,

    @Column(name = "question_id", nullable = false)
    var questionId: Long? = null,

    @Column(name = "answer", nullable = false)
    var answer: Char? = null
)
